pkg/login/social/github_oauth.go

```diff
@@ -201,9 +203,15 @@ func (s *SocialGithub) UserInfo(client *http.Client, token *oauth2.Token) (*Basi
 
 	teams := convertToGroupList(teamMemberships)
 
-	role, grafanaAdmin := s.extractRoleAndAdmin(response.Body, teams, true)
-	if s.roleAttributeStrict && !role.IsValid() {
-		return nil, &InvalidBasicRoleError{idP: "Github", assignedRole: string(role)}
+	var role roletype.RoleType
+	var grafanaAdmin bool
```

Comment:
I think currently you would always set Grafana server admin role to false if Github `skipOrgRoleSync` and `allowAssignGrafanaAdmin` are both enabled. It's a bit of a weird config setup, but I think we'd want to skip syncing server admin role in that case, right?

So you probably want to set `IsGrafanaAdmin` to `nil` if Github `skipOrgRoleSync` is enabled to avoid syncing that role. 

It would also be good to have tests covering server admin.

Reply:
Right, so even though we set server admin role to false we would want to set `IsGrafanaAdmin` to `nil`. 
https://github.com/grafana/grafana/blob/main/pkg/login/social/social.go#L237

Comments like these make my heart go ❤️ 

I'l add some tests covering this scenario. Should we also then set some debugging or maybe logs for stating that you have set both `allowAssignGrafanaAdmin` and `skipOrgRoleSync` where `skipOrgRoleSync` disables the assignment of GrafanaAdmin

Reply:
@IevaVasiljeva could you have another look? 🙏